#!/usr/bin/env python
PACKAGE = "aerial_robot_control"

from dynamic_reconfigure.parameter_generator_catkin import *
from aerial_robot_msgs.msg import DynamicReconfigureLevels

gen = ParameterGenerator()

gen.add("lqi_flag", bool_t, DynamicReconfigureLevels.RECONFIGURE_LQI_FLAG, "LQI Control Weight Flag", False)

gen.add("roll_pitch_p", double_t, DynamicReconfigureLevels.RECONFIGURE_LQI_ROLL_PITCH_P, "LQI Roll and Pitch P Control Weight", 200, 0, 2000)
gen.add("roll_pitch_i", double_t, DynamicReconfigureLevels.RECONFIGURE_LQI_ROLL_PITCH_I, "LQI Roll and Pitch I Control Weight", 10, 0, 50)
gen.add("roll_pitch_d", double_t, DynamicReconfigureLevels.RECONFIGURE_LQI_ROLL_PITCH_D, "LQI Roll and Pitch D Control Weight", 20, 0, 300)

gen.add("yaw_p", double_t, DynamicReconfigureLevels.RECONFIGURE_LQI_YAW_P, "LQI Yaw P Control Weight", 1, 0, 1000)
gen.add("yaw_i", double_t, DynamicReconfigureLevels.RECONFIGURE_LQI_YAW_I, "LQI Yaw I Control Weight", 1, 0, 20)
gen.add("yaw_d", double_t, DynamicReconfigureLevels.RECONFIGURE_LQI_YAW_D, "LQI Yaw D Control Weight", 10, 0, 1000)

gen.add("z_p", double_t, DynamicReconfigureLevels.RECONFIGURE_LQI_Z_P, "LQI Z P Control Weight", 1, -100, 100)
gen.add("z_i", double_t, DynamicReconfigureLevels.RECONFIGURE_LQI_Z_I, "LQI Z I Control Weight", 1, 0, 20)
gen.add("z_d", double_t, DynamicReconfigureLevels.RECONFIGURE_LQI_Z_D, "LQI Z D Control Weight", 10, -200, 200)

exit(gen.generate(PACKAGE, "aerial_robot_control", "LQI"))
